prereise/gather/demanddata/transportation_electrification/generate_scaling_factors.py

```diff
@@ -0,0 +1,206 @@
+    "29": "MO",
+    "30": "MT",
+    "31": "NE",
+    "32": "NV",
+    "33": "NH",
+    "34": "NJ",
+    "35": "NM",
+    "36": "NY",
+    "37": "NC",
+    "38": "ND",
+    "39": "OH",
+    "40": "OK",
+    "41": "OR",
+    "42": "PA",
+    "44": "RI",
+    "45": "SC",
+    "46": "SD",
+    "47": "TN",
+    "48": "TX",
+    "49": "UT",
+    "50": "VT",
+    "51": "VA",
+    "53": "WA",
+    "54": "WV",
+    "55": "WI",
+    "56": "WY",
+}
+
+
+def load_census_ua(path):
+    """Load census data for urban population
+
+    :param str path: path to census file (local or url)
+    :return: (*dict*) -- keys are state abbreviation and values are data frames giving
+        population by urban area in the state.
+    """
+    df = pd.read_excel(
+        path,
+        index_col=0,
+        keep_default_na=False,
+        usecols=[1, 2, 4],
+        skiprows=2,
+    )
+    state2ua = {}
+    for n, s in id2state.items():
+        state2ua[s] = df.query("STATE==@n")["POP"]
+
+    return state2ua
+
+
+def load_census_state(path):
+    """Load census data for state population
+
+    :param str path: path to census file (local or url)
+    :return: (*pandas.Series*) -- Indices are state abbreviations and values are
+        population in state
+    """
+    df = pd.read_excel(path, skiprows=3, index_col=0, skipfooter=7)
+    df.index = df.index.map(lambda x: str(x)[1:])
+
+    return df.loc[state2abv.keys()][2015].rename(index=state2abv)
+
+
+def load_dot_vmt_per_capita(path):
+    """Load Vehicle Miles Traveled (VMY) per capita in urban areas
+
+    :param str path: path to Department of Transportation's Transportation Health Tools
+        (local or url)
+    :return: (*tuple*) -- Series. Indices are state abbreviations and values are
+        population in ua (first element) or state (second element)
+    """
+    df_ua = pd.read_excel(
+        path,
+        sheet_name="Urbanized Area",
+        index_col=0,
+        usecols=[0, 3],
+        names=["UA", "VMT per Capita (daily)"],
+    )
+
+    df_state = pd.read_excel(
+        path,
+        sheet_name="State",
+        index_col=0,
+        usecols=[0, 39],
+        names=["State", "VMT per Capita (annual)"],
+    )
+
+    return (
+        df_ua.squeeze().loc[lambda x: x != "[no data]"],
+        df_state.rename(index=state2abv).squeeze(),
+    )
+
+
+def calculate_vmt_for_ua(census_ua, tht_ua):
+    """Calculate the total annual Vehicle Miles Traveled (VMT) in urban areas
+
+    :param dict census_ua: dictionary as returned by :func:`load_census_ua`
+    :param pandas.Series tht_ua: vmt per capita in urban areas as returned by
+        :func:`load_dot_vmt_per_capita`
+    :return: (*dict*) -- keys are state abbreviation and values are data frames giving
+        annual vmt by urban areas.
+    """
+
+    tht_ua_format = tht_ua.copy()
+    vmt_for_ua = {}
+
+    pattern = [",", "-", ".", " "]
+    for p in pattern:
+        tht_ua_format.index = tht_ua_format.index.str.replace(p, "", regex=True)
```

Comment:
Maybe we can do `tht_ua_format.index = tht_ua_format.index.str.replace(r'[, -.]', '', regex=True)`. Same comment below.

Reply:
Will try

Reply:
It works. Done